python/tank/bootstrap/manager.py

```diff
@@ -202,30 +208,42 @@ def _set_bundle_cache_fallback_paths(self, paths):
     )
 
 
-    def set_progress_callback(self, callback):
+    def set_progress_callback(self, progress_callback):
         """
-        Specify a method to call whenever progress should be reported back.
+        Sets a function to call whenever progress of the bootstrap should be reported back.
+
+        This function should have the following signature::
 
-        The method needs to have the following signature::
+            progress_callback(progress_value, message, current_index, maximum_index)
 
-            progress_callback(message, current_index, max_index)
+        where:
+        - ``progress_value`` is the current progress value, a float number ranging from 0.0 to 1.0
+                             representing the percentage of work completed.
+        - ``message`` is the progress message string to report.
+        - ``current_index`` is an optional current item number being looped over.
```

Comment:
Given that we now report total progress, so we need the `current_index` and `maximum_index`? Since we have already changed the interface of the callback, why not just simplify it down to 

```
progress_callback(progress_value, message)
```
Do you see a use case when current_index, maximum_index would be useful to have now that we have progress_value? To me it seems progress_value basically replaces current_index, maximum_index, so not sure why we would need all three!